modules/openxr/openxr_api.h

```diff
@@ -53,14 +53,25 @@
 // Note, OpenXR code that we wrote for our plugin makes use of C++20 notation for initialising structs which ensures zeroing out unspecified members.
 // Godot is currently restricted to C++17 which doesn't allow this notation. Make sure critical fields are set.
 
+// declare signals we use
+#define OPENXR_SESSION_BEGUN "session_begun"
+#define OPENXR_SESSION_STOPPING "session_stopping"
+#define OPENXR_SESSION_FOCUSSED "session_focussed"
+#define OPENXR_SESSION_VISIBLE "session_visible"
+#define OPENXR_POSE_RECENTERED "pose_recentered"
```

Comment:
You could possibly declare those with `SNAME()` to have statically allocated StringNames instead of coercing a string literal to a StringName each time.

And to do so it might make sense to move this to class constants in OpenXRInterface so they're local to that class.

Reply:
Looked at how it was applied in other classes and just followed suit, also instead of calling emit_signal on OpenXRInterface from OpenXRAPI, I've added nicer on_<status changed> functions to OpenXRInterface that are being called, and they then execute the emit signal, also means we can add additional logic needed on state change in OpenXRInterface of we choose to. I think that's a nicer solution.